packages/fetchai/skills/confirmation_aw3/registration_db.py

```diff
@@ -0,0 +1,141 @@
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains a the registration db model."""
+
+import datetime
+import json
+import logging
+import os
+import sqlite3
+from typing import Any, Dict, List, Tuple
+
+from aea.skills.base import Model
+
+
+_default_logger = logging.getLogger(
+    "aea.packages.fetchai.skills.confirmation_aw3.registration_db"
+)
+
+
+class RegistrationDB(Model):
+    """Communicate between the database and the python objects."""
+
+    def __init__(self, **kwargs):
+        """Initialise the class."""
+        custom_path = kwargs.pop("custom_path", None)
+        super().__init__(**kwargs)
+        this_dir = os.getcwd()
+        self.db_path = (
+            os.path.join(this_dir, "registration.db")
+            if custom_path is None
+            else custom_path
+        )
+        if not os.path.exists(os.path.dirname(os.path.abspath(self.db_path))):
+            raise ValueError(f"Path={self.db_path} not valid!")
+        self._initialise_backend()
+
+    def _initialise_backend(self) -> None:
+        """Set up database and initialise the tables."""
+        self._execute_single_sql(
+            "CREATE TABLE IF NOT EXISTS registered_table (address TEXT, ethereum_address TEXT, "
+            "ethereum_signature TEXT, fetchai_signature TEXT, "
+            "developer_handle TEXT, tweet TEXT)"
+        )
+        self._execute_single_sql(
+            "CREATE TABLE IF NOT EXISTS trades_table (address TEXT, created_at timestamp, data TEXT)"
+        )
+
+    def set_trade(
+        self, address: str, timestamp: datetime.datetime, data: Dict[str, str],
+    ) -> None:
+        """Record a registration."""
+        command = "INSERT INTO trades_table(address, created_at, data) values(?, ?, ?)"
+        variables: Tuple[str, datetime.datetime, str] = (
+            address,
+            timestamp,
+            json.dumps(data),
+        )
+        self._execute_single_sql(command, variables)
+
+    def get_trade_count(self, address: str) -> int:
+        """Get trade count."""
+        command = "SELECT COUNT(*) FROM trades_table where address=?"
+        ret = self._execute_single_sql(command, (address,))
+        return int(ret[0][0])
+
+    def get_developer_handle(self, address: str) -> str:
+        """Get developer handle."""
+        command = "SELECT developer_handle FROM registered_table where address=?"
+        ret = self._execute_single_sql(command, (address,))
+        return ret[0][0]
```

Comment:
why do you take the first? Since there isn't a primary key, it is not guaranteed to return just one result, maybe add a check if you don't want to change the schema?